### PR TITLE
Fix runtest.py for the dotnet/runtime repo

### PR DESCRIPTION
**Pull request by BruceForstall**
I simplified a lot of functions to just take the "args"
object, which contains a lot of paths and directories.
This makes it easier to centralize the directory computations
without spreading paths throughout the function signatures.

### CODE REVIEW

**Comment by BruceForstall** [comment]
Fixes https://github.com/dotnet/runtime/issues/645

**Review by jashook — APPROVED** [review]

File: src/coreclr/scripts/coreclr_arguments.py [hunk 1]
```diff
@@ -156,6 +158,20 @@ def verify(self,
     ############################################################################
 
     def __initialize__(self, args):
+
+        def provide_default_host_os():
```
Comment (sandreenko) [inline src/coreclr/scripts/coreclr_arguments.py]: what was the goal for moving these functions? Were any errors or do we have a python code convention?

**Review by sandreenko — COMMENTED** [review]
Thank you.

File: src/coreclr/scripts/coreclr_arguments.py (hunk 1 above)
Comment (BruceForstall) [reply src/coreclr/scripts/coreclr_arguments.py]: I thought it was confusing to have the callers appear before the callees. So, there's no real convention here that I knew of, but I did rearrange things to match my (personal) convention better... So you could say this change was somewhat arbitrary.

**Pull request merged**